saq/worker.py

```diff
@@ -307,22 +327,38 @@ async def process(self) -> bool:
             job.started = now()
             job.attempts += 1
             await job.update(status=Status.ACTIVE)
-            context = t.cast(CtxType, {**self.context, "job": job})
+            context = {**self.context, "job": job}
             await self._before_process(context)
             logger.info("Processing %s", job.info(logger.isEnabledFor(logging.DEBUG)))
 
-            function = ensure_coroutine_function(self.functions[job.function], self.pool)
+            function = self.functions.get(job.function)
+            assert function is not None
```

Comment:
why did you change this?

Reply:
My bad, not something I want in this PR. Now that the larger change looks agreeable to you let me ping you once the PR is ready for review.